### ISSUE TITLE
Entity as a type does not work

### ISSUE THREAD

**Issue by davydovanton** [hanami/model#410]
I think that "entity type" in entity attributes is a good idea and I want to use something like this:

```ruby
class Post < Hanami::Entity
  attributes do
    attribute :id,         Types::Int
    attribute :created_at, Types::Time
    attribute :updated_at, Types::Time
    attribute :deleted_at, Types::Time
    attribute :name,       Types::String
    attribute :body,       Types::String
    attribute :comments,   Types::Collection(Comment)
    attribute :user,       User
  end
end
```

But this code does not work. For fix this issue I need to add a hack like this:
```ruby
class Post < Hanami::Entity
  attributes do
    attribute :id,         Types::Int
    attribute :created_at, Types::Time
    attribute :updated_at, Types::Time
    attribute :deleted_at, Types::Time
    attribute :name,       Types::String
    attribute :body,       Types::String
    attribute :comments,   Types::Collection(Comment)
    attribute :user,       Hash
  end

  def user
    User.new(super)
  end
end
```

I think we can create a new type (like `Type::Entity(User)`) or update entity for using this as a type.

WDYT?

/cc @hanami/core

**Comment by jodosha** [comment]
@davydovanton @mereghost what do we use for _one-to-many_ relationship?

**Comment by davydovanton** [comment]
@jodosha 
```ruby
    attribute :users, Types::Collection(User)
```

**Comment by mereghost** [comment]
On the many-to-one (and one-to-one) side of things the type is initialized as such:

```ruby
Sql::Types::Schema::AssociationType.new(entity)
```

Which itself is a subclass of `Hanami::Model::Types::Schema::CoercibleType`

**Comment by jodosha** [comment]
@davydovanton I was referring to the other side of the association, which is what @mereghost mentioned.

@mereghost Is it possible to rename it to `Types::Entity`, so it can be used like:

```ruby
attribute :user, Types::Entity(User)
```

What do you think?

**Comment by mereghost** [comment]
I think it could work since it would be a simple class rename. :P

**Comment by jodosha** [comment]
@mereghost It's all yours, then please take care of this.

**Comment by jodosha** [comment]
@mereghost Please make this happen 💚

**Comment by jodosha** [comment]
Implemented by #457
